dspace-api/src/main/java/org/dspace/deletion/process/DSpaceObjectDeletionProcess.java

```diff
@@ -0,0 +1,164 @@
+import org.dspace.scripts.DSpaceRunnable;
+import org.dspace.utils.DSpace;
+
+/**
+ * Batch process for deleting DSpace objects (Item, Collection, Community).
+ * This class orchestrates the deletion process, delegating the actual deletion logic
+ * to a strategy registry that selects the appropriate strategy for each DSpaceObject type.
+ *
+ * @author Mykhaylo Boychuk (mykhaylo.boychuk@4science.com)
+ */
+public class DSpaceObjectDeletionProcess
+        extends DSpaceRunnable<DSpaceObjectDeletionProcessScriptConfiguration<DSpaceObjectDeletionProcess>> {
+
+    private ItemService itemService;
+    private HandleService handleService;
+    private CommunityService communityService;
+    private AuthorizeService authorizeService;
+    private CollectionService collectionService;
+
+    private String id;
+    private Context context;
+    private String[] copyVirtualMetadata;
+    private List<DSpaceObjectDeletionStrategy> deletionStrategies = new ArrayList<>();
+
+    @Override
+    public void setup() throws ParseException {
+        ServiceManager serviceManager = new DSpace().getServiceManager();
+        itemService = ContentServiceFactory.getInstance().getItemService();
+        handleService = HandleServiceFactory.getInstance().getHandleService();
+        communityService = ContentServiceFactory.getInstance().getCommunityService();
+        collectionService = ContentServiceFactory.getInstance().getCollectionService();
+        authorizeService = AuthorizeServiceFactory.getInstance().getAuthorizeService();
+        deletionStrategies.addAll(serviceManager.getServicesByType(DSpaceObjectDeletionStrategy.class));
+
+        parseCommandLineOptions();
+    }
+
+    /**
+     * Parses and validates command line options.
+     */
+    private void parseCommandLineOptions() {
+        this.id = commandLine.getOptionValue('i');
+        this.copyVirtualMetadata = commandLine.hasOption('c') ? parseCopyVirtualMetadataOption() : new String[0];
+    }
+
+    private String[] parseCopyVirtualMetadataOption() {
+        String value = commandLine.getOptionValue('c');
+        if (value.contains(",")) {
+            return value.split(",");
+        }
+        return new String[] { value };
+    }
+
+    @Override
+    public void internalRun() throws Exception {
+        assignCurrentUserInContext();
+        Optional<DSpaceObject> dSpaceObjectOptional = resolveDSpaceObject(this.id);
+
+        if (dSpaceObjectOptional.isEmpty()) {
+            var error = String.format("DSpaceObject for provided identifier:%s doesn't exist!", this.id);
+            throw new IllegalArgumentException(error);
+        }
+
+        DSpaceObject dso = dSpaceObjectOptional.get();
+
+        if (!authorizeService.isAdmin(context, dso)) {
+            throw new AuthorizeException("Current user is not eligible to execute script 'dspace-object-deletion'");
+        }
+
+        handler.logInfo("Performing deletion of DSpaceObject id:" + dso.getID());
```

Comment:
I'd recommend enhancing this logged message so that it also lists the type and notes that child objects are deleted.  Maybe something like this:

`"Performing deletion of DSpaceObject (and all child objects) for type=" + Constants.typeText[dso.getType()] + " and id=" + dso.getID()`